packages/jest-mock/src/__tests__/index.test.ts

```diff
@@ -2432,3 +2440,26 @@ test('`fn` and `spyOn` do not throw', () => {
     spyOn({apple: () => {}}, 'apple');
   }).not.toThrow();
 });
+
+describe('Explicit Resource Management', () => {
+  it('jest.fn state should be restored with the `using` keyword', () => {
+    const mock = jest.fn();
+    {
+      using inScope = mock.mockReturnValue(2);
+      expect(inScope()).toBe(2);
+      expect(mock()).toBe(2);
+    }
+    expect(mock()).not.toBe(2);
+  });
+
+  it('should be restored with the `using` keyword', () => {
+    {
+      using mockedLog = jest.spyOn(console, 'log');
+      expect(jest.isMockFunction(console.log)).toBeTruthy();
+
+      console.log('test');
+      expect(mockedLog).toHaveBeenCalled();
+    }
+    expect(jest.isMockFunction(console.log)).toBeFalsy();
```

Comment:
This is a common usage:

`using mockedLog = jest.spyOn(console, 'log');` is only mocked inside of the current scope, and as soon as the block is left, the mock is restored.

Reply:
this is awesome!